app/components/lunch-invites/invite/status.js

```diff
@@ -2,18 +2,22 @@ import Ember from 'ember';
 
 export default Ember.Component.extend({
   profiles: Ember.inject.service(),
-  tagName: 'td',
-  classNames: ['grey'],
-  classNameBindings: ['unreadMessages'],
+  tagName: '',
   unreadMessages: null,
 
   onInit: function() {
     const id = this.get('conversation.id');
-    return this.get('profiles').getMessages(id)
+    return this.get('profiles').getMessagesForConversation(id)
```

Comment:
By doing this you remove that extra `then` and you can clearly see what the code is trying to do.